pitaya-sharp/NPitaya/src/Metrics/MetricsReporter.cs

```diff
@@ -0,0 +1,103 @@
+            // TODO (felipe.rodopoulos): prometheus-net does not support subsystem label yet. We'll use a hardcoded one.
+            var name = Marshal.PtrToStringAnsi(opts.Name);
+            var prometheusReporter = RetrievePrometheus(prometheusPtr);
+            var key = BuildKey(name);
+            prometheusReporter.RegisterCounter(key);
+        }
+
+        static void RegisterHistogramFn(IntPtr prometheusPtr, MetricsOpts opts)
+        {
+            var ns = Marshal.PtrToStringAnsi(opts.MetricNamespace);
+            // TODO (felipe.rodopoulos): prometheus-net does not support subsystem label yet. We'll use a hardcoded one.
+            var name = Marshal.PtrToStringAnsi(opts.Name);
+            var prometheusReporter = RetrievePrometheus(prometheusPtr);
+            var key = BuildKey(name);
+            prometheusReporter.RegisterHistogram(key);
+        }
+
+        static void RegisterGaugeFn(IntPtr prometheusPtr, MetricsOpts opts)
+        {
+            var ns = Marshal.PtrToStringAnsi(opts.MetricNamespace);
+            // TODO (felipe.rodopoulos): prometheus-net does not support subsystem label yet. We'll use a hardcoded one.
+            var name = Marshal.PtrToStringAnsi(opts.Name);
+            var prometheusReporter = RetrievePrometheus(prometheusPtr);
+            var key = BuildKey(name);
+            prometheusReporter.RegisterGauge(key);
+        }
+
+        static void IncCounterFn(IntPtr prometheusPtr, IntPtr name, ref IntPtr labels, UInt32 labelsCount)
+        {
+            string nameStr = Marshal.PtrToStringAnsi(name);
+            var prometheus = RetrievePrometheus(prometheusPtr);
+            prometheus.IncCounter(nameStr);
+        }
+
+        static void ObserveHistFn(IntPtr prometheusPtr, IntPtr name, double value, ref IntPtr labels, UInt32 labelsCount)
+        {
+            string nameStr = Marshal.PtrToStringAnsi(name);
+            var key = BuildKey(nameStr);
+            var prometheus = RetrievePrometheus(prometheusPtr);
+            prometheus.ObserveHistogram(key, value);
+        }
+
+        static void SetGaugeFn(IntPtr prometheusPtr, IntPtr name, double value, ref IntPtr labels, UInt32 labelsCount)
+        {
+            string nameStr = Marshal.PtrToStringAnsi(name);
+            var prometheus = RetrievePrometheus(prometheusPtr);
+            prometheus.SetGauge(nameStr, value);
+        }
+
+        static void AddGaugeFn(IntPtr prometheusPtr, IntPtr name, double value, ref IntPtr labels, UInt32 labelsCount)
+        {
+            string nameStr = Marshal.PtrToStringAnsi(name);
+            Logger.Warn($"Adding gauge {nameStr} with val {value}. This method should not be used.");
+        }
+
+        private static PrometheusReporter RetrievePrometheus(IntPtr ptr)
+        {
+            var handle = GCHandle.FromIntPtr(ptr);
+            return handle.Target as PrometheusReporter;
+        }
+
+        private static string BuildKey(string suffix)
```

Comment:
consider avoiding the `private` keyword, since it is not used on the rest of the codebase.

Reply:
Can you explain why, exactly? Seems a good practice to use private in this context.

Reply:
by default everything is private.

Reply:
Oh! Good to know!